openfl/component/envoy/envoy.py

```diff
@@ -73,26 +82,74 @@ def send_health_check(self):
             )
             time.sleep(DEFAULT_TIMEOUT_IN_SECONDS / 2)
 
-    def _run_collaborator(self, experiment_name, plan='plan/plan.yaml',):
+    def _run_collaborator(self, *, experiment_work_dir: Path):
         """Run the collaborator for the experiment running."""
-        cwd = os.getcwd()
-        os.chdir(f'{cwd}/{experiment_name}')  # TODO: probably it should be another way
-
         # This is needed for python module finder
-        sys.path.append(os.getcwd())
+        sys.path.append(str(experiment_work_dir))
 
-        plan = Plan.parse(plan_config_path=Path(plan))
+        plan = Plan.parse(
+            plan_config_path=Path('plan/plan.yaml'),
+            working_dir=experiment_work_dir
+        )
 
         # TODO: Need to restructure data loader config file loader
         echo(f'Data = {plan.cols_data_paths}')
         logger.info('🧿 Starting a Collaborator Service.')
 
-        col = plan.get_collaborator(self.name, self.root_ca, self.key,
-                                    self.cert, shard_descriptor=self.shard_descriptor)
+        col = plan.get_collaborator(
+            self.name,
+            self.root_ca,
+            self.key,
+            self.cert,
+            shard_descriptor=self.shard_descriptor
+        )
         try:
             col.run()
         finally:
-            os.chdir(cwd)
+            if experiment_work_dir in sys.path:
+                sys.path.remove(str(experiment_work_dir))
+
+    @staticmethod
+    def remove_workspace(experiment_name):
+        """Remove the workspace."""
+        shutil.rmtree(experiment_name, ignore_errors=True)
+
+    @staticmethod
+    def create_workspace(experiment_name, experiment_work_dir, response_iter):
```

Comment:
Again, we have this functionality in three places: Director, Envoy, and the old CLI's `workspace import`.
We may think of creating one helper function and use it everywhere.